### PR TITLE
tbl_merge() allows tbl_summary() merging

### PR DESCRIPTION
**Pull request by ddsjoberg**
#230 
This update now allows `tbl_summary()` objects to be merged alongside regression objects.  The change required a rather large update to the handling of footnotes.  

Updates made:
- Footnotes are now stored in the `.$table_header` object.  At the end of `tbl_summary()`, `tbl_regression()`, and `tbl_uvregression()` (and the related functions) rebuild the {gt} footnote code at the end of each function.  
- There are 2 types of footnotes: Abbreviations, and regular footnotes, and there is a column in `.$table_header` for each type
    - Abbreviations are consolidated, and reported as a single footnote.  These include abbreviation definitions like Odds Ratio and Confidence Interval.
    - Regular footnotes are more straightforward.  Each footnote listed in this column results in a footnote in the table.
- In `tbl_merge()`, the merging of the tables has been limited to columns common among `tbl_summary()`, `tbl_regression()`, and `tbl_uvregression()`.  This is the largest change in this function, and likely has the largest chance of introducing bugs. Previously we merged on `var_type` as well (a regression object column).  My primary concern is that some non-standard regression merges will have broken.
- There were also some fixes related to the removal of `cols_merge()` in a previous dev release.
- The way headers for logistic, Poisson, Cox, etc are identified has been improved.

### CODE REVIEW

**Comment by ddsjoberg** [comment]
@michaelcurry1123 thank you for reviewing this massive PR!!! :)

**Comment by michaelcurry1123** [comment]
first thing tomorrow :)

**Comment by ddsjoberg** [comment]
Updates to make:
1. Allow merging of two or more `tbl_summary` objects
1. Move the footnote for `tbl_summary` objects to the first column beginning with `stat_`

File: R/tbl_merge.R [hunk 1]
```diff
@@ -1,13 +1,14 @@
 #' Merge two or more gtsummary regression objects
 #'
-#' Merges two or more `tbl_regression`, `tbl_uvregression`, or `tbl_stack`
-#' objects and adds appropriate spanning headers.
+#' Merges two or more `tbl_regression`, `tbl_uvregression`, `tbl_stack`,
+#' of `tbl_summary` objects and adds appropriate spanning headers.
```
Comment (margarethannum) [inline R/tbl_merge.R]: replace "of" with "or"

Comment (ddsjoberg) [reply R/tbl_merge.R]: 🙈 updated! 🙏

**Comment by margarethannum** [comment]
Is it problematic that when there is missing data (and that missing data is excluded from a model) - if you merge it with the summary information, there's nowhere for the model N to go? 

``` r
library(survival)
library(gtsummary)
library(dplyr)
#> 
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union

ts <- trial %>%
  dplyr::select(age, grade, trt) %>%
  tbl_summary()
t1 <-
  glm(response ~ trt + grade + age, trial, family = binomial) %>%
  tbl_regression(exponentiate = TRUE)

t1
```



<style>html {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Helvetica Neue', 'Fira Sans', 'Droid Sans', Arial, sans-serif;
}

#oudvvbkvni .gt_table {
  display: table;
  border-collapse: collapse;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  background-color: #FFFFFF;
  /* table.background.color */
  width: auto;
  /* table.width */
  border-top-style: solid;
  /* table.border.top.style */
  border-top-width: 2px;
  /* table.border.top.width */
  border-top-color: #A8A8A8;
  /* table.border.top.color */
  border-bottom-style: solid;
  /* table.border.bottom.style */
  border-bottom-width: 2px;
  /* table.border.bottom.width */
  border-bottom-color: #A8A8A8;
  /* table.border.bottom.color */
}

#oudvvbkvni .gt_heading {
  background-color: #FFFFFF;
  /* heading.background.color */
  border-bottom-color: #FFFFFF;
}

#oudvvbkvni .gt_title {
  color: #333333;
  font-size: 125%;
  /* heading.title.font.size */
  padding-top: 4px;
  /* heading.top.padding - not yet used */
  padding-bottom: 4px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#oudvvbkvni .gt_subtitle {
  color: #333333;
  font-size: 85%;
  /* heading.subtitle.font.size */
  padding-top: 0;
  padding-bottom: 4px;
  /* heading.bottom.padding - not yet used */
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#oudvvbkvni .gt_bottom_border {
  border-bottom-style: solid;
  /* heading.border.bottom.style */
  border-bottom-width: 2px;
  /* heading.border.bottom.width */
  border-bottom-color: #D3D3D3;
  /* heading.border.bottom.color */
}

#oudvvbkvni .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  padding-top: 4px;
  padding-bottom: 4px;
}

#oudvvbkvni .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  /* column_labels.background.color */
  font-size: 16px;
  /* column_labels.font.size */
  font-weight: initial;
  /* column_labels.font.weight */
  vertical-align: middle;
  padding: 5px;
  margin: 10px;
  overflow-x: hidden;
}

#oudvvbkvni .gt_columns_top_border {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
}

#oudvvbkvni .gt_columns_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#oudvvbkvni .gt_sep_right {
  border-right: 5px solid #FFFFFF;
}

#oudvvbkvni .gt_group_heading {
  padding: 8px;
  /* row_group.padding */
  color: #333333;
  background-color: #FFFFFF;
  /* row_group.background.color */
  font-size: 16px;
  /* row_group.font.size */
  font-weight: initial;
  /* row_group.font.weight */
  border-top-style: solid;
  /* row_group.border.top.style */
  border-top-width: 2px;
  /* row_group.border.top.width */
  border-top-color: #D3D3D3;
  /* row_group.border.top.color */
  border-bottom-style: solid;
  /* row_group.border.bottom.style */
  border-bottom-width: 2px;
  /* row_group.border.bottom.width */
  border-bottom-color: #D3D3D3;
  /* row_group.border.bottom.color */
  vertical-align: middle;
}

#oudvvbkvni .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  /* row_group.background.color */
  font-size: 16px;
  /* row_group.font.size */
  font-weight: initial;
  /* row_group.font.weight */
  border-top-style: solid;
  /* row_group.border.top.style */
  border-top-width: 2px;
  /* row_group.border.top.width */
  border-top-color: #D3D3D3;
  /* row_group.border.top.color */
  border-bottom-style: solid;
  /* row_group.border.bottom.style */
  border-bottom-width: 2px;
  /* row_group.border.bottom.width */
  border-bottom-color: #D3D3D3;
  /* row_group.border.bottom.color */
  vertical-align: middle;
}

#oudvvbkvni .gt_striped {
  background-color: #8080800D;
}

#oudvvbkvni .gt_from_md > :first-child {
  margin-top: 0;
}

#oudvvbkvni .gt_from_md > :last-child {
  margin-bottom: 0;
}

#oudvvbkvni .gt_row {
  padding: 8px;
  /* row.padding */
  margin: 10px;
  border-bottom-style: solid;
  border-bottom-width: thin;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#oudvvbkvni .gt_stub {
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 12px;
}

#oudvvbkvni .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  /* summary_row.background.color */
  padding: 8px;
  /* summary_row.padding */
  text-transform: inherit;
  /* summary_row.text_transform */
}

#oudvvbkvni .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  /* grand_summary_row.background.color */
  padding: 8px;
  /* grand_summary_row.padding */
  text-transform: inherit;
  /* grand_summary_row.text_transform */
}

#oudvvbkvni .gt_first_summary_row {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
}

#oudvvbkvni .gt_first_grand_summary_row {
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#oudvvbkvni .gt_table_body {
  border-top-style: solid;
  /* table_body.border.top.style */
  border-top-width: 2px;
  /* table_body.border.top.width */
  border-top-color: #D3D3D3;
  /* table_body.border.top.color */
  border-bottom-style: solid;
  /* table_body.border.bottom.style */
  border-bottom-width: 2px;
  /* table_body.border.bottom.width */
  border-bottom-color: #D3D3D3;
  /* table_body.border.bottom.color */
}

#oudvvbkvni .gt_footnotes {
  border-top-style: solid;
  /* footnotes.border.top.style */
  border-top-width: 2px;
  /* footnotes.border.top.width */
  border-top-color: #D3D3D3;
  /* footnotes.border.top.color */
}

#oudvvbkvni .gt_footnote {
  font-size: 90%;
  /* footnote.font.size */
  margin: 0px;
  padding: 4px;
  /* footnote.padding */
}

#oudvvbkvni .gt_sourcenotes {
  border-top-style: solid;
  /* sourcenotes.border.top.style */
  border-top-width: 2px;
  /* sourcenotes.border.top.width */
  border-top-color: #D3D3D3;
  /* sourcenotes.border.top.color */
}

#oudvvbkvni .gt_sourcenote {
  font-size: 90%;
  /* sourcenote.font.size */
  padding: 4px;
  /* sourcenote.padding */
}

#oudvvbkvni .gt_center {
  text-align: center;
}

#oudvvbkvni .gt_left {
  text-align: left;
}

#oudvvbkvni .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#oudvvbkvni .gt_font_normal {
  font-weight: normal;
}

#oudvvbkvni .gt_font_bold {
  font-weight: bold;
}

#oudvvbkvni .gt_font_italic {
  font-style: italic;
}

#oudvvbkvni .gt_super {
  font-size: 65%;
}

#oudvvbkvni .gt_footnote_marks {
  font-style: italic;
  font-size: 65%;
}
</style>

<div id="oudvvbkvni" style="overflow-x:auto;overflow-y:auto;width:auto;height:auto;">

<table class="gt_table">

<tr>

<th class="gt_col_heading gt_columns_bottom_border gt_columns_top_border gt_left" rowspan="1" colspan="1">

<strong>N = 182</strong>

</th>

<th class="gt_col_heading gt_columns_bottom_border gt_columns_top_border gt_center" rowspan="1" colspan="1">

<strong>OR</strong><sup class="gt_footnote_marks">1</sup>

</th>

<th class="gt_col_heading gt_columns_bottom_border gt_columns_top_border gt_center" rowspan="1" colspan="1">

<strong>95% CI</strong><sup class="gt_footnote_marks">1</sup>

</th>

<th class="gt_col_heading gt_columns_bottom_border gt_columns_top_border gt_center" rowspan="1" colspan="1">

<strong>p-value</strong>

</th>

</tr>

<body class="gt_table_body">

<tr>

<td class="gt_row gt_left">

Treatment Randomization

</td>

<td class="gt_row gt_center">

</td>

<td class="gt_row gt_center">

</td>

<td class="gt_row gt_center">

</td>

</tr>

<tr>

<td class="gt_row gt_left gt_striped" style="text-align: left; text-indent: 10px;">

Drug

</td>

<td class="gt_row gt_center gt_striped">

—

</td>

<td class="gt_row gt_center gt_striped">

—

</td>

<td class="gt_row gt_center gt_striped">

</td>

</tr>

<tr>

<td class="gt_row gt_left" style="text-align: left; text-indent: 10px;">

Placebo

</td>

<td class="gt_row gt_center">

0.52

</td>

<td class="gt_row gt_center">

0.28, 0.94

</td>

<td class="gt_row gt_center">

0.032

</td>

</tr>

<tr>

<td class="gt_row gt_left gt_striped">

Grade

</td>

<td class="gt_row gt_center gt_striped">

</td>

<td class="gt_row gt_center gt_striped">

</td>

<td class="gt_row gt_center gt_striped">

</td>

</tr>

<tr>

<td class="gt_row gt_left" style="text-align: left; text-indent: 10px;">

I

</td>

<td class="gt_row gt_center">

—

</td>

<td class="gt_row gt_center">

—

</td>

<td class="gt_row gt_center">

</td>

</tr>

<tr>

<td class="gt_row gt_left gt_striped" style="text-align: left; text-indent: 10px;">

II

</td>

<td class="gt_row gt_center gt_striped">

0.60

</td>

<td class="gt_row gt_center gt_striped">

0.28, 1.29

</td>

<td class="gt_row gt_center gt_striped">

0.2

</td>

</tr>

<tr>

<td class="gt_row gt_left" style="text-align: left; text-indent: 10px;">

III

</td>

<td class="gt_row gt_center">

0.85

</td>

<td class="gt_row gt_center">

0.41, 1.74

</td>

<td class="gt_row gt_center">

0.6

</td>

</tr>

<tr>

<td class="gt_row gt_left gt_striped">

Age, yrs

</td>

<td class="gt_row gt_center gt_striped">

1.00

</td>

<td class="gt_row gt_center gt_striped">

0.98, 1.02

</td>

<td class="gt_row gt_center gt_striped">

0.7

</td>

</tr>

</body>

<tfoot>

<tr class="gt_footnotes">

<td colspan="4">

<p class="gt_footnote">

<sup class="gt_footnote_marks">
<em>1</em>
</sup>

OR = Odds Ratio, CI = Confidence Interval
<br />

</p>

</td>

</tr>

</tfoot>

</table>

</div>



``` r

tbl_merge(
  tbls = list(ts, t1), 
  tab_spanner = c("Summary", "Tumor Response")
)
```



<style>html {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Helvetica Neue', 'Fira Sans', 'Droid Sans', Arial, sans-serif;
}

#fnuyqxbbpe .gt_table {
  display: table;
  border-collapse: collapse;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  background-color: #FFFFFF;
  /* table.background.color */
  width: auto;
  /* table.width */
  border-top-style: solid;
  /* table.border.top.style */
  border-top-width: 2px;
  /* table.border.top.width */
  border-top-color: #A8A8A8;
  /* table.border.top.color */
  border-bottom-style: solid;
  /* table.border.bottom.style */
  border-bottom-width: 2px;
  /* table.border.bottom.width */
  border-bottom-color: #A8A8A8;
  /* table.border.bottom.color */
}

#fnuyqxbbpe .gt_heading {
  background-color: #FFFFFF;
  /* heading.background.color */
  border-bottom-color: #FFFFFF;
}

#fnuyqxbbpe .gt_title {
  color: #333333;
  font-size: 125%;
  /* heading.title.font.size */
  padding-top: 4px;
  /* heading.top.padding - not yet used */
  padding-bottom: 4px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#fnuyqxbbpe .gt_subtitle {
  color: #333333;
  font-size: 85%;
  /* heading.subtitle.font.size */
  padding-top: 0;
  padding-bottom: 4px;
  /* heading.bottom.padding - not yet used */
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#fnuyqxbbpe .gt_bottom_border {
  border-bottom-style: solid;
  /* heading.border.bottom.style */
  border-bottom-width: 2px;
  /* heading.border.bottom.width */
  border-bottom-color: #D3D3D3;
  /* heading.border.bottom.color */
}

#fnuyqxbbpe .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  padding-top: 4px;
  padding-bottom: 4px;
}

#fnuyqxbbpe .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  /* column_labels.background.color */
  font-size: 16px;
  /* column_labels.font.size */
  font-weight: initial;
  /* column_labels.font.weight */
  vertical-align: middle;
  padding: 5px;
  margin: 10px;
  overflow-x: hidden;
}

#fnuyqxbbpe .gt_columns_top_border {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
}

#fnuyqxbbpe .gt_columns_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#fnuyqxbbpe .gt_sep_right {
  border-right: 5px solid #FFFFFF;
}

#fnuyqxbbpe .gt_group_heading {
  padding: 8px;
  /* row_group.padding */
  color: #333333;
  background-color: #FFFFFF;
  /* row_group.background.color */
  font-size: 16px;
  /* row_group.font.size */
  font-weight: initial;
  /* row_group.font.weight */
  border-top-style: solid;
  /* row_group.border.top.style */
  border-top-width: 2px;
  /* row_group.border.top.width */
  border-top-color: #D3D3D3;
  /* row_group.border.top.color */
  border-bottom-style: solid;
  /* row_group.border.bottom.style */
  border-bottom-width: 2px;
  /* row_group.border.bottom.width */
  border-bottom-color: #D3D3D3;
  /* row_group.border.bottom.color */
  vertical-align: middle;
}

#fnuyqxbbpe .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  /* row_group.background.color */
  font-size: 16px;
  /* row_group.font.size */
  font-weight: initial;
  /* row_group.font.weight */
  border-top-style: solid;
  /* row_group.border.top.style */
  border-top-width: 2px;
  /* row_group.border.top.width */
  border-top-color: #D3D3D3;
  /* row_group.border.top.color */
  border-bottom-style: solid;
  /* row_group.border.bottom.style */
  border-bottom-width: 2px;
  /* row_group.border.bottom.width */
  border-bottom-color: #D3D3D3;
  /* row_group.border.bottom.color */
  vertical-align: middle;
}

#fnuyqxbbpe .gt_striped {
  background-color: #8080800D;
}

#fnuyqxbbpe .gt_from_md > :first-child {
  margin-top: 0;
}

#fnuyqxbbpe .gt_from_md > :last-child {
  margin-bottom: 0;
}

#fnuyqxbbpe .gt_row {
  padding: 8px;
  /* row.padding */
  margin: 10px;
  border-bottom-style: solid;
  border-bottom-width: thin;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#fnuyqxbbpe .gt_stub {
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 12px;
}

#fnuyqxbbpe .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  /* summary_row.background.color */
  padding: 8px;
  /* summary_row.padding */
  text-transform: inherit;
  /* summary_row.text_transform */
}

#fnuyqxbbpe .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  /* grand_summary_row.background.color */
  padding: 8px;
  /* grand_summary_row.padding */
  text-transform: inherit;
  /* grand_summary_row.text_transform */
}

#fnuyqxbbpe .gt_first_summary_row {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
}

#fnuyqxbbpe .gt_first_grand_summary_row {
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#fnuyqxbbpe .gt_table_body {
  border-top-style: solid;
  /* table_body.border.top.style */
  border-top-width: 2px;
  /* table_body.border.top.width */
  border-top-color: #D3D3D3;
  /* table_body.border.top.color */
  border-bottom-style: solid;
  /* table_body.border.bottom.style */
  border-bottom-width: 2px;
  /* table_body.border.bottom.width */
  border-bottom-color: #D3D3D3;
  /* table_body.border.bottom.color */
}

#fnuyqxbbpe .gt_footnotes {
  border-top-style: solid;
  /* footnotes.border.top.style */
  border-top-width: 2px;
  /* footnotes.border.top.width */
  border-top-color: #D3D3D3;
  /* footnotes.border.top.color */
}

#fnuyqxbbpe .gt_footnote {
  font-size: 90%;
  /* footnote.font.size */
  margin: 0px;
  padding: 4px;
  /* footnote.padding */
}

#fnuyqxbbpe .gt_sourcenotes {
  border-top-style: solid;
  /* sourcenotes.border.top.style */
  border-top-width: 2px;
  /* sourcenotes.border.top.width */
  border-top-color: #D3D3D3;
  /* sourcenotes.border.top.color */
}

#fnuyqxbbpe .gt_sourcenote {
  font-size: 90%;
  /* sourcenote.font.size */
  padding: 4px;
  /* sourcenote.padding */
}

#fnuyqxbbpe .gt_center {
  text-align: center;
}

#fnuyqxbbpe .gt_left {
  text-align: left;
}

#fnuyqxbbpe .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#fnuyqxbbpe .gt_font_normal {
  font-weight: normal;
}

#fnuyqxbbpe .gt_font_bold {
  font-weight: bold;
}

#fnuyqxbbpe .gt_font_italic {
  font-style: italic;
}

#fnuyqxbbpe .gt_super {
  font-size: 65%;
}

#fnuyqxbbpe .gt_footnote_marks {
  font-style: italic;
  font-size: 65%;
}
</style>

<div id="fnuyqxbbpe" style="overflow-x:auto;overflow-y:auto;width:auto;height:auto;">

<table class="gt_table">

<tr>

<th class="gt_col_heading gt_center gt_columns_bottom_border" rowspan="2" colspan="1">

<strong>Characteristic</strong>

</th>

<th class="gt_col_heading gt_center gt_columns_top_border gt_column_spanner gt_sep_right" rowspan="1" colspan="1">

Summary

</th>

<th class="gt_col_heading gt_center gt_columns_top_border gt_column_spanner" rowspan="1" colspan="3">

Tumor Response

</th>

</tr>

<tr>

<th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1">

<strong>N = 200</strong><sup class="gt_footnote_marks">1</sup>

</th>

<th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1">

<strong>OR</strong><sup class="gt_footnote_marks">2</sup>

</th>

<th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1">

<strong>95% CI</strong><sup class="gt_footnote_marks">2</sup>

</th>

<th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1">

<strong>p-value</strong>

</th>

</tr>

<body class="gt_table_body">

<tr>

<td class="gt_row gt_left">

Age, yrs

</td>

<td class="gt_row gt_center">

46 (37, 57)

</td>

<td class="gt_row gt_center">

1.00

</td>

<td class="gt_row gt_center">

0.98, 1.02

</td>

<td class="gt_row gt_center">

0.7

</td>

</tr>

<tr>

<td class="gt_row gt_left gt_striped" style="text-align: left; text-indent: 10px;">

Unknown

</td>

<td class="gt_row gt_center gt_striped">

9

</td>

<td class="gt_row gt_center gt_striped">

</td>

<td class="gt_row gt_center gt_striped">

</td>

<td class="gt_row gt_center gt_striped">

</td>

</tr>

<tr>

<td class="gt_row gt_left">

Grade

</td>

<td class="gt_row gt_center">

</td>

<td class="gt_row gt_center">

</td>

<td class="gt_row gt_center">

</td>

<td class="gt_row gt_center">

</td>

</tr>

<tr>

<td class="gt_row gt_left gt_striped" style="text-align: left; text-indent: 10px;">

I

</td>

<td class="gt_row gt_center gt_striped">

67 (34%)

</td>

<td class="gt_row gt_center gt_striped">

—

</td>

<td class="gt_row gt_center gt_striped">

—

</td>

<td class="gt_row gt_center gt_striped">

</td>

</tr>

<tr>

<td class="gt_row gt_left" style="text-align: left; text-indent: 10px;">

II

</td>

<td class="gt_row gt_center">

58 (29%)

</td>

<td class="gt_row gt_center">

0.60

</td>

<td class="gt_row gt_center">

0.28, 1.29

</td>

<td class="gt_row gt_center">

0.2

</td>

</tr>

<tr>

<td class="gt_row gt_left gt_striped" style="text-align: left; text-indent: 10px;">

III

</td>

<td class="gt_row gt_center gt_striped">

75 (38%)

</td>

<td class="gt_row gt_center gt_striped">

0.85

</td>

<td class="gt_row gt_center gt_striped">

0.41, 1.74

</td>

<td class="gt_row gt_center gt_striped">

0.6

</td>

</tr>

<tr>

<td class="gt_row gt_left">

Treatment Randomization

</td>

<td class="gt_row gt_center">

</td>

<td class="gt_row gt_center">

</td>

<td class="gt_row gt_center">

</td>

<td class="gt_row gt_center">

</td>

</tr>

<tr>

<td class="gt_row gt_left gt_striped" style="text-align: left; text-indent: 10px;">

Drug

</td>

<td class="gt_row gt_center gt_striped">

107 (54%)

</td>

<td class="gt_row gt_center gt_striped">

—

</td>

<td class="gt_row gt_center gt_striped">

—

</td>

<td class="gt_row gt_center gt_striped">

</td>

</tr>

<tr>

<td class="gt_row gt_left" style="text-align: left; text-indent: 10px;">

Placebo

</td>

<td class="gt_row gt_center">

93 (46%)

</td>

<td class="gt_row gt_center">

0.52

</td>

<td class="gt_row gt_center">

0.28, 0.94

</td>

<td class="gt_row gt_center">

0.032

</td>

</tr>

</body>

<tfoot>

<tr class="gt_footnotes">

<td colspan="5">

<p class="gt_footnote">

<sup class="gt_footnote_marks">
<em>1</em>
</sup>

Statistics presented: median (IQR); n (%)
<br />

</p>

<p class="gt_footnote">

<sup class="gt_footnote_marks">
<em>2</em>
</sup>

OR = Odds Ratio, CI = Confidence Interval
<br />

</p>

</td>

</tr>

</tfoot>

</table>

</div>



<sup>Created on 2019-10-03 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by margarethannum** [comment]
When you merge a `tbl_summary` object from something with missing data to the corresponding `tbl_regression` object (that excluded the missing data in the model), there's currently no indication of the N used in the model. But I don't have a good suggestion on where the model N should go! 

for example, in the below, there are 9 missing data points, but when the merge occurs all you see is the "N = 200" in the summary column which could be misleading since the regression only used N = 182. 

image

```r 
library(survival)
library(gtsummary)
library(dplyr)

ts <- trial %>%
  dplyr::select(age, grade, trt) %>%
  tbl_summary()
t1 <-
  glm(response ~ trt + grade + age, trial, family = binomial) %>%
  tbl_regression(exponentiate = TRUE)

t1

tbl_merge(
  tbls = list(ts, t1), 
  tab_spanner = c("Summary", "Tumor Response")
)
```

**Comment by margarethannum** [comment]
Similar non-intuitive N labeling occurs if you merge two models that used a different sample size - only the N from the first model is displayed. 

eg.

``` r
library(survival)
library(gtsummary)

# make dataset that is arbitrarily different size
trial2 <- trial %>%
  filter(age > 30)

t1 <-
  glm(response ~ trt + grade + age, trial, family = binomial) %>%
  tbl_regression(exponentiate = TRUE)

# build model with different sample size
t2 <-
  coxph(Surv(ttdeath, death) ~ trt + grade + age, trial2) %>%
  tbl_regression(exponentiate = TRUE)

t2

tbl_merge(
    tbls = list(t1, t2),
    tab_spanner = c("Tumor Response", "Time to Death")
  )
```
image

image

**Comment by margarethannum** [comment]
Maybe it's a problem people should be aware of before they merge tables (it's not necessarily {gtsummary}'s job to make sure people are doing the right analysis), but maybe there could be a check and message if you are merging objects with a different N?

**Comment by michaelcurry1123** [comment]
some interesting merges in specific cases. in the code below if you try to merge where the label is different for `grade` you will see it creates two variables (as expected because it merges on label and name) but interestingly the level that is the same across the variable appears in the wrong section. the best way to handle this is to make sure the labels are the same. I happens for the models as well. 


`library(tidyverse)
library(gtsummary)

trial2 <- mutate(trial, grade = as.character(grade),
                 grade = ifelse(grade == "I","I","II+"))
trial <- mutate(trial, grade = as.character(grade))

Hmisc::label(trial2$grade) <- "Grade"
Hmisc::label(trial$grade) <- "grade"

sumobj1 <- tbl_summary(trial)

sumobj2 <- tbl_summary(trial2)


tbl_merge(list(sumobj1,sumobj2))



mod1 <- tbl_regression(glm(marker ~ grade + trt, trial, 
                           family = gaussian))

mod2 <- tbl_regression(glm(marker ~ grade + trt, trial2, family = gaussian))


tbl_merge(list(mod1,mod2))`

**Comment by michaelcurry1123** [comment]
same thing here I was expecting unknown to be appear the the categorical group and the continuous group - I think what I am posting is minor but wanted to show you

`trial2 <- mutate(trial, grade = as.character(grade),
                 grade = ifelse(grade == "I","I","II+"),
                 age = ifelse(age <= median(age,na.rm = TRUE), "younger","older"))
trial <- mutate(trial, grade = as.character(grade))

Hmisc::label(trial2$age) <- "Age cat"


sumobj1 <- tbl_summary(trial)

sumobj2 <- tbl_summary(trial2)


tbl_merge(list(sumobj1,sumobj2))`

**Comment by ddsjoberg** [comment]
@margarethannum and I discussed the N issue and we decided on the following changes:
- [ ] Update `tbl_regression()` to no longer report the N in the column label.  All merged tables will have default label "**Characteristic**"
- [ ] Need to update the documentation for `tbl_regression()` omitting the part about the N presented.
- [ ] Add a note in the `tbl_merge()` that Ns should be checked.

@michaelcurry1123 oy, great catch!  I think if I merge on variable name, variable label, AND label (the column that is being presented), it'll fix the problem.
- [ ] Update `tbl_merge()` to also merge on the variable label

**Comment by margarethannum** [comment]
@ddsjoberg

**Comment by margarethannum** [comment]
Sorry about unintentional closing/reopening. @ddsjoberg I agree, no note necessary if we're not including the N from one table as default anymore in the tbl_merge().

**Comment by michaelcurry1123** [comment]
@ddsjoberg yea I think you are correct. happy to test it when you update it !

**Comment by ddsjoberg** [comment]
@michaelcurry1123 @margarethannum 
Thank you for this amazing review!  I think everything should be updated per our conversation now.

**Pull request merged**